### DISCUSSION TITLE
GitHub Actions Workflow Security

### DISCUSSION THREAD

**Discussion by jardizzo** [community/community#19632] [Actions]
We have a public repo with on-prem CI workflows. We identified two ways to limit execution of the sensitive workflows: (1) use of labels and (2) use of environments. However, (1) and (2) are useless if access to the workflow files cannot be constrained. We do not want anyone other than the repo owner and administrators to have access to the workflow files. Is that possilble? Any help is greatly appreciated. Reusable workflows from private repos is an open discussion that could also be a potential solution.

**Comment by daltonv** [comment]
Is forking not an option? You could change your workflow so commits from the public are only accepted from forked versions of the repo.  If the users always fork into their own namespace they won't have access to the organizations self-hosted runners.

You might be able to set up some forking rules with the new features github recently deployed https://github.blog/changelog/2022-06-27-improved-innersource-collaboration-and-enterprise-fork-policies/

**Reply by jardizzo** [reply]
Thanks so much for the suggestion. We are currently reviewing this as an option and will post an update.
